sdk/trace/span.go

```diff
@@ -265,17 +276,44 @@ func (s *recordingSpan) addOverCapAttrs(attrs []attribute.KeyValue) {
 			continue
 		}
 
-		if len(s.attributes) >= s.tracer.provider.spanLimits.AttributeCountLimit {
+		if len(s.attributes) >= limit {
 			// Do not just drop all of the remaining attributes, make sure
 			// updates are checked and performed.
 			s.droppedAttributes++
 		} else {
+			truncateAttr(s.tracer.provider.spanLimits.AttributeValueLengthLimit, &a)
 			s.attributes = append(s.attributes, a)
 			exists[a.Key] = len(s.attributes) - 1
 		}
 	}
 }
 
+// truncateAttr trucates the value of attr. Only string and string slice
+// attribute values are truncated. String values are truncated to at most a
+// length of limit. String slice values have each slice value truncated in
+// this fasion (the slice length itself is unaffected).
+//
+// No truncation is perfromed for a negative limit.
+func truncateAttr(limit int, attr *attribute.KeyValue) {
+	if limit < 0 {
+		return
+	}
+	switch attr.Value.Type() {
+	case attribute.STRING:
+		if v := attr.Value.AsString(); len(v) > limit {
+			attr.Value = attribute.StringValue(v[:limit])
+		}
+	case attribute.STRINGSLICE:
+		v := attr.Value.AsStringSlice()
+		for i := range v {
+			if len(v[i]) > limit {
+				v[i] = v[i][:limit]
+			}
+		}
+		// No need to update attr, underlying slice array was updated.
+	}
```

Comment:
Is this appropriate?  I thought that Attributes were supposed to be read-only after creation, so that if they happen to be shared that one couldn't be bashed by a different process.

Reply:
Sure I'll update.

Reply:
Updated, PTAL.